core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -239,96 +240,186 @@ void testStartup() {
         assertEquals(logDir, capture.getValue().get("log.dir"));
     }
 
-    // This test creates 2 log segments, 1st one has start offset of 0, 2nd one (and active one) has start offset of 150.
```

Comment:
To make the review easier: in this test file,
- `testCopyLogSegmentsToRemoteShouldCopyExpectedLogSegment` was moved to a test subclass;
- The code that initializes the log mock was extracted to `setUp`;
- A new test `testCustomMetadataSizeExceedsLimit` was added.

Reply:
Thanks for the note.